### ISSUE TITLE
Edit Dataverse Drop Down: Broken/Odd Behavior

### ISSUE THREAD

**Issue by eaquigley** [IQSS/dataverse#1679]
Was setting up some stuff in dataverse-demo and discovered when I click on published in the facets on the homepage, then click on Edit Dataverse->General Information this happens:

![brokendropdown](https://cloud.githubusercontent.com/assets/5922904/6659984/716b59de-cb66-11e4-9130-8b8a23ccf49c.jpg)

**Comment by pdurbin** [comment]
The bug seems to happen whenever to you nothing in the id (rather than 1 or some other number) like this:

http://localhost:8080/dataverse.xhtml?id=

I checked in a potential fix at 25ed0b1

Without a fix like this, the stacktrace looks like this:

```
[2015-03-16T08:23:13.430-0400] [glassfish 4.1] [WARNING] [] [javax.enterprise.resource.webcontainer.jsf.lifecycle] [tid: _ThreadID=105 _ThreadName=http-listener-1(3)] [timeMillis: 1426508593430] [levelValue: 900] [[
  /dataverse.xhtml @358,157 rendered="#{dataverseSession.user.authenticated and                                                      (permissionServiceBean.on(DataversePage.dataverse).canIssueCommand('UpdateDataverseCommand')                                                         or permissionServiceBean.on(DataversePage.dataverse).canIssueCommand('PublishDataverseCommand'))}": javax.ejb.EJBException
javax.el.ELException: /dataverse.xhtml @358,157 rendered="#{dataverseSession.user.authenticated and                                                      (permissionServiceBean.on(DataversePage.dataverse).canIssueCommand('UpdateDataverseCommand')                                                         or permissionServiceBean.on(DataversePage.dataverse).canIssueCommand('PublishDataverseCommand'))}": javax.ejb.EJBException
	at com.sun.faces.facelets.el.TagValueExpression.getValue(TagValueExpression.java:114)
	at javax.faces.component.ComponentStateHelper.eval(ComponentStateHelper.java:194)
	at javax.faces.component.UIComponentBase.isRendered(UIComponentBase.java:457)
	at javax.faces.component.UIComponentBase.processDecodes(UIComponentBase.java:1210)
	at javax.faces.component.UIComponentBase.processDecodes(UIComponentBase.java:1220)
	at javax.faces.component.UIForm.processDecodes(UIForm.java:225)
	at javax.faces.component.UIComponentBase.processDecodes(UIComponentBase.java:1220)
	at javax.faces.component.UIComponentBase.processDecodes(UIComponentBase.java:1220)
	at javax.faces.component.UIViewRoot.processDecodes(UIViewRoot.java:929)
	at com.sun.faces.lifecycle.ApplyRequestValuesPhase.execute(ApplyRequestValuesPhase.java:78)
	at com.sun.faces.lifecycle.Phase.doPhase(Phase.java:101)
	at com.sun.faces.lifecycle.LifecycleImpl.execute(LifecycleImpl.java:198)
	at javax.faces.webapp.FacesServlet.service(FacesServlet.java:646)
	at org.apache.catalina.core.StandardWrapper.service(StandardWrapper.java:1682)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:344)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:214)
	at org.glassfish.tyrus.servlet.TyrusServletFilter.doFilter(TyrusServletFilter.java:295)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:256)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:214)
	at org.ocpsoft.rewrite.servlet.RewriteFilter.doFilter(RewriteFilter.java:205)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:256)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:214)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:316)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:160)
	at org.apache.catalina.core.StandardPipeline.doInvoke(StandardPipeline.java:734)
	at org.apache.catalina.core.StandardPipeline.invoke(StandardPipeline.java:673)
	at com.sun.enterprise.web.WebPipeline.invoke(WebPipeline.java:99)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:174)
	at org.apache.catalina.connector.CoyoteAdapter.doService(CoyoteAdapter.java:415)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:282)
	at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:459)
	at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:167)
	at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:201)
	at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:175)
	at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:235)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:561)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:112)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:117)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:56)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:137)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:565)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:545)
	at java.lang.Thread.run(Thread.java:745)
Caused by: javax.el.ELException: javax.ejb.EJBException
	at javax.el.ELUtil.invokeMethod(ELUtil.java:336)
	at javax.el.BeanELResolver.invoke(BeanELResolver.java:537)
	at javax.el.CompositeELResolver.invoke(CompositeELResolver.java:256)
	at com.sun.el.parser.AstValue.getValue(AstValue.java:136)
	at com.sun.el.parser.AstValue.getValue(AstValue.java:204)
	at com.sun.el.parser.AstOr.getValue(AstOr.java:58)
	at com.sun.el.parser.AstAnd.getValue(AstAnd.java:63)
	at com.sun.el.ValueExpressionImpl.getValue(ValueExpressionImpl.java:226)
	at org.jboss.weld.el.WeldValueExpression.getValue(WeldValueExpression.java:50)
	at com.sun.faces.facelets.el.TagValueExpression.getValue(TagValueExpression.java:109)
	... 48 more
Caused by: javax.ejb.EJBException
	at com.sun.ejb.containers.EJBContainerTransactionManager.processSystemException(EJBContainerTransactionManager.java:748)
	at com.sun.ejb.containers.EJBContainerTransactionManager.completeNewTx(EJBContainerTransactionManager.java:698)
	at com.sun.ejb.containers.EJBContainerTransactionManager.postInvokeTx(EJBContainerTransactionManager.java:503)
	at com.sun.ejb.containers.BaseContainer.postInvokeTx(BaseContainer.java:4566)
	at com.sun.ejb.containers.BaseContainer.postInvoke(BaseContainer.java:2074)
	at com.sun.ejb.containers.BaseContainer.postInvoke(BaseContainer.java:2044)
	at com.sun.ejb.containers.EJBLocalObjectInvocationHandler.invoke(EJBLocalObjectInvocationHandler.java:220)
	at com.sun.ejb.containers.EJBLocalObjectInvocationHandlerDelegate.invoke(EJBLocalObjectInvocationHandlerDelegate.java:88)
	at com.sun.proxy.$Proxy307.on(Unknown Source)
	at edu.harvard.iq.dataverse.__EJB31_Generated__PermissionServiceBean__Intf____Bean__.on(Unknown Source)
	at sun.reflect.GeneratedMethodAccessor183.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.jboss.weld.util.reflection.Reflections.invokeAndUnwrap(Reflections.java:414)
	at org.jboss.weld.bean.proxy.EnterpriseBeanProxyMethodHandler.invoke(EnterpriseBeanProxyMethodHandler.java:127)
	at org.jboss.weld.bean.proxy.EnterpriseTargetBeanInstance.invoke(EnterpriseTargetBeanInstance.java:56)
	at org.jboss.weld.bean.proxy.InjectionPointPropagatingEnterpriseTargetBeanInstance.invoke(InjectionPointPropagatingEnterpriseTargetBeanInstance.java:65)
	at org.jboss.weld.bean.proxy.ProxyMethodHandler.invoke(ProxyMethodHandler.java:100)
	at edu.harvard.iq.dataverse.PermissionServiceBean$Proxy$_$$_Weld$EnterpriseProxy$.on(Unknown Source)
	at sun.reflect.GeneratedMethodAccessor182.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at javax.el.ELUtil.invokeMethod(ELUtil.java:332)
	... 57 more
Caused by: java.lang.IllegalArgumentException: Cannot query permissions on a DvObject with a null id.
	at edu.harvard.iq.dataverse.PermissionServiceBean.on(PermissionServiceBean.java:250)
```
```

**Comment by pdurbin** [comment]
@scolapasta please move this ticket to QA if you think it's ready.

I just deployed this fix to https://apitest.dataverse.org and I can't reproduce the problem there anymore. @eaquigley you are welcome to use this server but please give me a heads up because I tend to drop the database there at will.

**Comment by scolapasta** [comment]
I think a better fix would not be to have the id as part of the URL in these links - you'll notice at the subdataverse it doesn't need it either any more since the URL now has the alias, for example, the links are:

http://localhost:8080/dataverse/gpd?id=4&q=&fq0=published_ss%3A%22Unpublished%22&types=dataverses%3Adatasets&sort=release_or_create_date_dt&order=desc

The id gets ignored since the alias is there and should not be part of the URL any more.

This also works for the root without id (I tried the following link with your change commented out):
http://localhost:8080/?q=&fq0=published_ss%3A%22Unpublished%22&types=dataverses%3Adatasets&sort=release_or_create_date_dt&order=desc

This might not be that important, but I don't love the id of leaving the URL in these links where it doesn;t have to be (I am aware there are other links with ids still in them, but they would require back end code to change at this point, that I'm not sure we can get to)

(I think it does make sense to leave your line of code in too, just in case we somehow leave id there somewhere - no reason not to set the value)

Thoughts?

**Comment by pdurbin** [comment]
Right. This code is so old we didn't have aliases for dataverse, just id numbers.

We already have "Shorten and clean up bookmarkable search URLs (and refactor duplicate code)" at #264 and I was planning on removing the id numbers then, I guess. But it's a non-trivial amount of work and that ticket is slated for a future milestone. I say we give this one to QA as is for beta 15.

**Comment by pdurbin** [comment]
Discussed with @scolapasta 

I'll see if I think pulling out all the `id`'s will break anything.

**Comment by pdurbin** [comment]
@scolapasta I think you're right. That old dataverseId parameter (a `long`) is legacy at this point. I removed it completely (Java and xhtml) 1a2fbc6

A good clue that it's not (apparently) necessary is that dataverseuser.xhtml had no reference to it. dataverse.xhtml did on the other hand, so I removed it.

Heads up to @mheppler that it was in iframe.xhtml but apparently not used. That file seems to use "alias" instead.

Now the only place it's referenced is in sample-search-include-page.xhtml and sample-account-page-with-search.xhtml which I'll remove now per #1610

Passing to QA. Clicking facets and what not should no longer add an "id" to the URL.

For QA,
